src/solve.rs

```diff
@@ -224,6 +225,14 @@ fn build_payload_for_swap_and_approval(
             value: swap.value,
             call_data: swap.data.0.clone(),
             exec_plan: None,
+            inputs: vec![TokenAmount {
+                token: query.sell_token,
+                amount: swap.sell_amount,
```

Comment:
Sometimes, for buy orders, the 0x api is buying too many tokens and then also receiving more tokens than what it got. But it seems that this is also not always happening. The logic will be a little more intrigued. I will do some more investigations and then prepare a solution in a separate PR. For sell-orders, this logic will always work.